File: Selection_Sort/selection_sort.cpp
/* Program for selection sort. */

#include<bits/stdc++.h>
using namespace std;

/* Selection sort function. */
void selection_sort(vector<int> &A)
{
	/* Size of the vector. */
	int n = A.size();

	/* Loop for the starting index of the unsorted part. */
	for (int i = 0; i < n; ++i)
	{
		/* variable to store the least element of the
		unsorted part and index of that element. */
		int minm = 1e6, in = -1;

		/* Loop for finding the minimum value in the sorted part. */
		for (int j = i; j < n; ++j)
		{
			if (A[j] < minm)
			{
				minm = A[j];
				in = j;
			}
		}

		/* Swapping the minimum value with the first element of the unsorted part. */
		swap(A[in], A[i]);
	}
}


int main()
{

	/* Size of Vector. */
	int n;
	cin >> n;

	vector<int> A(n);

	/* Inputing the values. */
	for (int i = 0; i < n; i++) cin >> A[i];

	/* Passing A as referance to selection_sort and updating it.*/
	selection_sort(A);

	/* Printing the final sorted vector. */
	for (int i = 0; i < n; i++) cout << A[i] << " ";
	cout << endl;

	return 0;
}

File: Selection_Sort/selection_sort.py
#funtion for selection sort
def selection_sort(A):
    # Traverse through all array elements 
    for i in range(len(A)): 
      
        # Find the minimum element in remaining unsorted array 
        min_idx=i 
        for j in range(i+1, len(A)): 
            if A[min_idx]>A[j]: 
                min_idx=j 
              
        # Swap the found minimum element with the first element         
        A[i], A[min_idx]=A[min_idx], A[i] 
n=int(input("Enter number of element:"))
array=[]
print("Enter array:")
for i in range(0,n):
    e=input()
    #adding element to array
    array.append(e)

#funtion call
selection_sort(array)

print("Sorted array:") 
for i in range(len(array)): 
    print(array[i]) 
